Snowflake.Data/Core/Session/SFSessionProperty.cs

```diff
@@ -117,8 +117,13 @@ class SFSessionProperties : Dictionary<SFSessionProperty, String>
                 SFSessionProperty.PRIVATE_KEY_PWD,
                 SFSessionProperty.PROXYPASSWORD,
             };
-        
-        private const string AccountRegexString = "^\\w[\\w.-]+\\w$";
+
+        private static readonly List<string> s_accountRegexStrings = new List<string>

```

Comment:
why not just single
`^[\w.-]+$`

Reply:
it is too general- works for example 'ss_dd_dd_dd_dd.d..d' .  I'm thinking about ^\w+(.|-)?\w+$` but still underscore is in grup represented by \w. The underscore is a correct separator?

Reply:
@sfc-gh-mhofman it would allow the account name to start or finish with `.` or `-`

Reply:
@sfc-gh-pmotacki what you are thinking doesn't solve the problem. For example 1 letter account would match to the regex

Reply:
Agreed on this version